src/main/java/com/mongodb/kafka/connect/source/heartbeat/HeartbeatManager.java

```diff
@@ -61,36 +61,43 @@ public HeartbeatManager(
 
   public Optional<SourceRecord> heartbeat() {
     if (cursor == null) {
+      LOGGER.debug("Returning no heartbeat: null cursor");
       return Optional.empty();
     }
-
     long currentMS = time.milliseconds();
     long timeSinceHeartbeatMS = currentMS - lastHeartbeatMS;
-
-    if (canCreateHeartbeat && timeSinceHeartbeatMS > heartbeatIntervalMS) {
-      lastHeartbeatMS = currentMS;
-      return Optional.ofNullable(cursor.getResumeToken())
-          .map(
-              r -> {
-                String resumeToken = r.toJson();
-                if (!resumeToken.equals(lastResumeToken)) {
-                  LOGGER.info("Generating heartbeat event. {}", resumeToken);
-                  Map<String, String> sourceOffset = new HashMap<>();
-                  sourceOffset.put(ID_FIELD, resumeToken);
-                  sourceOffset.put(HEARTBEAT_KEY, "true");
-                  lastResumeToken = resumeToken;
-                  return new SourceRecord(
-                      partitionMap,
-                      sourceOffset,
-                      heartbeatTopicName,
-                      Schema.STRING_SCHEMA,
-                      resumeToken,
-                      Schema.OPTIONAL_BYTES_SCHEMA,
-                      null);
-                }
-                return null;
-              });
+    if (!canCreateHeartbeat) {
+      LOGGER.debug("Returning no heartbeat: canCreateHeartbeat is false");
```

Comment:
Can we instead log the reason that `canCreateHeartbeat` is false?  It's not clear what the semantics of this variable is based just on its name.

Reply:
This couples the message to the canCreateHeartbeat logic, which is in a different method (the constructor). But it seems like it should be in this method, since this is where the rest of the checks are. I moved it.